code/modules/integrated_electronics/subtypes/input.dm

```diff
@@ -431,3 +633,83 @@
 	activate_pin(1)
 	return TRUE
 
+/obj/item/integrated_circuit/input/internalbm
+	name = "internal battery monitor"
+	desc = "This monitors the charge level of an internal battery."
+	icon_state = "internalbm"
+	extended_desc = "This circuit will give you values of charge, max charge and percentage of the internal battery on demand."
+	w_class = ITEMSIZE_TINY
+	complexity = 1
+	inputs = list()
+	outputs = list(
+		"cell charge" = IC_PINTYPE_NUMBER,
+		"max charge" = IC_PINTYPE_NUMBER,
+		"percentage" = IC_PINTYPE_NUMBER,
+		"refference to assembly" = IC_PINTYPE_REF
+		)
+	activators = list("read" = IC_PINTYPE_PULSE_IN, "on read" = IC_PINTYPE_PULSE_OUT)
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+	origin_tech = list(TECH_ENGINEERING = 4, TECH_DATA = 4, TECH_POWER = 4, TECH_MAGNET = 3)
+	power_draw_per_use = 1
+
+/obj/item/integrated_circuit/input/internalbm/do_work()
+	set_pin_data(IC_OUTPUT, 1, null)
+	set_pin_data(IC_OUTPUT, 2, null)
+	set_pin_data(IC_OUTPUT, 3, null)
+	set_pin_data(IC_OUTPUT, 4, weakref(assembly))
+	if(assembly)
+		if(assembly.battery)
+
+			set_pin_data(IC_OUTPUT, 1, assembly.battery.charge)
+			set_pin_data(IC_OUTPUT, 2, assembly.battery.maxcharge)
+			set_pin_data(IC_OUTPUT, 3, 100*assembly.battery.charge/assembly.battery.maxcharge)
+	push_data()
+	activate_pin(2)
+
+/obj/item/integrated_circuit/input/externalbm
+	name = "external battery monitor"
+	desc = "This can help to watch battery state of any device in view"
```

Comment:
"This can help watch the battery level of any device in range."

Reply:
I changed, but this device can read data only in line of sight.